qiskit_optimization/applications/max_k_cut.py

```diff
@@ -0,0 +1,198 @@
+            k: The number of colors
+            colors: List of strings or list of colors in rgba lists to be assigned to each
+                resulting subset, there must be as many colors as the number k
+        """
+        super().__init__(graph=graph)
+        self._subsets_num = k
+        self._colors = colors if colors and len(colors) >= k else None
+
+    def to_quadratic_program(self) -> QuadraticProgram:
+        """Convert a Max-k-cut problem instance into a
+        :class:`~qiskit_optimization.problems.QuadraticProgram`
+
+        Returns:
+            The :class:`~qiskit_optimization.problems.QuadraticProgram` created
+            from the Max-k-cut problem instance.
+        """
+        for w, v in self._graph.edges:
+            self._graph.edges[w, v].setdefault("weight", 1)
+
+        mdl = Model(name="Max-k-cut")
+        n = self._graph.number_of_nodes()
+        k = self._subsets_num
+        x = {(v, i): mdl.binary_var(name=f"x_{v}_{i}") for v in range(n) for i in range(k)}
+        first_penalty = mdl.sum_squares((1 - mdl.sum(x[v, i] for i in range(k)) for v in range(n)))
+        second_penalty = mdl.sum(
+            mdl.sum(self._graph.edges[v, w]["weight"] * x[v, i] * x[w, i] for i in range(k))
+            for v, w in self._graph.edges
+        )
+        objective = first_penalty + second_penalty
+        mdl.minimize(objective)
+
+        op = from_docplex_mp(mdl)
+        return op
+
+    def interpret(self, result: Union[OptimizationResult, np.ndarray]) -> List[List[int]]:
+        """Interpret a result as k lists of node indices
+
+        Args:
+            result : The calculated result of the problem
+
+        Returns:
+            k lists of node indices correspond to k node sets for the Max-k-cut
+        """
+        x = self._result_to_x(result)
+        n = self._graph.number_of_nodes()
+        cut = [[] for i in range(self._subsets_num)]  # type: List[List[int]]
+
+        n_selected = x.reshape((n, self._subsets_num))
+        for i in range(n):
+            node_in_subset = np.where(n_selected[i] == 1)[0]  # one-hot encoding
+            if len(node_in_subset) != 0:
+                cut[node_in_subset[0]].append(i)
+
+        return cut
+
+    def _draw_result(
+        self,
+        result: Union[OptimizationResult, np.ndarray],
+        pos: Optional[Dict[int, np.ndarray]] = None,
+    ) -> None:
+        """Draw the result with colors
+
+        Args:
+            result : The calculated result for the problem
+            pos: The positions of nodes
+        """
+        x = self._result_to_x(result)
+        nx.draw(self._graph, node_color=self._node_color(x), pos=pos, with_labels=True)
+
+    def _node_color(
+        self, x: np.ndarray
+    ) -> Union[List[Tuple[float, float, float, float]], List[str]]:
+        # Return a list of colors for draw.
+
+        n = self._graph.number_of_nodes()
+
+        # k colors chosen (randomly or from cm.rainbow), or from given color list
```

Comment:
As far as doing things randomly there exists an random generator in algorithm_globals in Terra that we created to have a common generator that algorithms and applications would use so that a user could create reproducible results by seeding it. Using that, instead of whats done here, would allow a user to seed it and have the same colored drawing each time if they prefer - of course they could pass in their own colors too, but I think using the common generator is preferable.

Reply:
Now `algorithm_globals` is used